plugins/org.python.pydev.ast/src/org/python/pydev/ast/interpreter_managers/InterpreterInfo.java

```diff
@@ -1622,6 +1624,10 @@ public void setEnvVariables(String[] env) {
             return env; //nothing to change
         }
 
+        if (!this.condaEnv.isEmpty()) {
```

Comment:
You should first put in a local var and then check for `isEmpty` to avoid racing conditions and then use that same variable in the createEnvWithMap.